### PR TITLE
Check capabilities before generating plot reprs

### PR DESCRIPTION
**Pull request by takluyver**
Closes IRkernel/IRkernel#96, hopefully.

### CODE REVIEW

File: R/recordedplot.r [hunk 1]
```diff
@@ -27,6 +27,7 @@ repr_png.recordedplot <- function(p,
 	#special
 	res       = getOption('repr.plot.res'),
 ...) {
+	if (!capabilities('png')) { return NULL }
```
Comment (takluyver) [inline R/recordedplot.r]: I assume this being FALSE would mean we couldn't generate a png - likewise for jpeg below.

Comment (flying-sheep) [inline R/recordedplot.r]: yeah, although i can’t imagine what setup would result in this being false… embedded R? :laughing:

yeah, you can configure it to build without but i doubt anyone using repr would ever do this. still, why not put the check in?

**Comment by flying-sheep** [comment]
won’t work. try `if (!capabilities('png')) return(NULL)`

**Comment by takluyver** [comment]
Thanks, fixed.

We should work on some automated tests for this, but that can be after the release. I'll file a separate issue.

**Pull request merged**